iroh-net/src/discovery/static_provider.rs

```diff
@@ -0,0 +1,162 @@
+//! A static discovery implementation that allows adding info for nodes manually.
```

Comment:
This should really explain **why** this thing exists.  In what situations do you want to use this?  Maybe this should include an example using a Ticket even.  It could suggest that you may want to remove items when you know they're stale for some reason.  etc.

All the same things also need to be on `StaticProvider` directly, so it's fine to only write this down once and refer to the other place for the other.